driver-core/src/main/com/mongodb/TaggableReadPreference.java

```diff
@@ -186,14 +188,14 @@ protected static ClusterDescription copyClusterDescription(final ClusterDescript
 
         long heartbeatFrequencyMS = clusterDescription.getServerSettings().getHeartbeatFrequency(MILLISECONDS);
 
-        if (getMaxStaleness(MILLISECONDS) < Math.max(SMALLEST_MAX_STALENESS_MS, heartbeatFrequencyMS + IDLE_WRITE_PERIOD_MS)) {
+        if (maxStaleness < Math.max(SMALLEST_MAX_STALENESS_MS, heartbeatFrequencyMS + IDLE_WRITE_PERIOD_MS)) {
```

Comment:
Replace method call with Nullable return value with a local variable.  This is a common theme, as the static analyzer doesn't "know" that the method will return the same value for every call and that therefore a single check for null at the start of the method is sufficient.  

This is a common theme in this PR, which I don't particularly like, but don't see a way around.  It could be solved with additional annotations (e.g. @Immutable) but that would require changes in tools that we don't control (SpotBugs, IntelliJ)